java/core/src/main/java/com/whylogs/core/metrics/Metric.java

```diff
@@ -0,0 +1,48 @@
+package com.whylogs.core.metrics;
+
+import com.whylogs.core.PreProcessedColumn;
+import com.whylogs.core.SummaryConfig;
+
+import java.util.HashMap;
+
+public abstract class Metric{
+
+    private String namespace;
+
+    // TODO: needs a __post_init__ for the registry
+
+    public static String getNameSpace(Metric metric, MetricConfig config){
+        return metric.zero(config).namespace;
+    }
+
+    public abstract <T extends Metric> T zero(MetricConfig config);
```

Comment:
I wonder if we could just make this a ctor? I don't think we need this to be generic.

Reply:
Or it needs to be static and generic...

Reply:
I don't think we do either. Will update this when I update the other section of metric